cql3/statements/cf_prop_defs.hh

```diff
@@ -89,11 +89,13 @@ public:
     static constexpr int32_t DEFAULT_MAX_INDEX_INTERVAL = 2048;
 private:
     mutable std::optional<sstables::compaction_strategy_type> _compaction_strategy_class;
+    mutable std::optional<schema::extensions_map> _schema_extensions;
```

Comment:
Nit: This is more caching of schema_extension than lazy initialization. This patch does not change the time schema extensions will be created. I understand why you're doing that but current commit message is a bit confusing.

Reply:
I updated the commit description to name it as "caching", not "lazy initialization".